triggers/daml/Daml/Trigger/LowLevel.daml

```diff
@@ -26,17 +30,27 @@ module Daml.Trigger.LowLevel
 
 import DA.Next.Map (MapKey(..))
 
-data Identifier = Identifier
-  { packageId : Text
-  , moduleName : Text
-  , entityName : Text
-  } deriving (Show, Eq, Ord)
-
 data AnyContractId = AnyContractId
-  { templateId : Identifier
+  { templateId : TemplateTypeRep
   , contractId : Text
   } deriving (Show, Eq, Ord)
 
+-- XXX: The name ContractId clashes with Prelude.ContractId.
+--   Find a better name than TheContractId.
+newtype TheContractId t = TheContractId { contractId : Text }
```

Comment:
Hm, not quite sure if having to hide Prelude is better or worse. Slightly leaning towards just requiring hiding Prelude and/or qualified imports but neither is great.

Maybe we can come up with some terrible hack to get to and from text functions for absolute contract ids and then just use the actual ContractId type but I’m not quite sure how to pull that off.

Reply:
It's tricky, requiring to hide `Prelude.ContractId` increases the boilerplate for creating triggers. Also, trigger modules might still require `Prelude.ContractId`, e.g. [here](https://github.com/digital-asset/daml/blob/6c36777a8ee65e0942233d3bea02bc29b4e583d9/triggers/tests/daml/ACS.daml#L91), so using the same name can be confusing.

Reply:
Maybe `AbsoluteContractId` isn’t so bad? The normal `ContractId` can be both relative (while building up a transaction) and absolute which is the part that stops us from implementing conversions to `Text` so this at least conveys the right distinction for why we can do things here that we can’t do with the normal `ContractId` type.

Reply:
Yes, that sounds better. I've changed it to that for now.